File: CvTracker.Client/models/JobOffer.ts
import type { ApplicationStatus } from './ApplicationStatus'
import type { JobOfferNote } from './JobOfferNote'

export interface JobOffer {
  id: number
  position: string
  contractType: string
  workMode: string
  workLoad: string
  companyName: string | null
  location: string | null
  requiredSkillIds: number[]
  requiredSkillNames: string[]
  requiredSkillLevels: Record<number, string>
  requiredSkills?: Array<{ technologyId: number; requiredLevel: string }>
  matchScore: number | null
  sourceUrl?: string | null
  status: ApplicationStatus
  salaryMin: number | null
  salaryMax: number | null
  appliedAt: string | null
  followUpDate: string | null
  recruiterName: string | null
  recruiterContact: string | null
  sentCvVersion: string | null
  rejectionReason: string | null
  notes: JobOfferNote[]
}

File: CvTracker.Api/Controllers/Models/JobOffer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CvTracker.Api.Models;

public class JobOffer
{
    public int Id { get; set; }
    public required string Position { get; set; }

    public ContractType ContractType { get; set; }  // UoP / B2B / ...
    public WorkLoad WorkLoad { get; set; }           // FullTime / PartTime
    public WorkMode WorkMode { get; set; }           // Remote / OnSite / Hybrid

    public string? CompanyName { get; set; }
    public string? Location { get; set; }

    [NotMapped] public int? MatchScore { get; set; }

    [Url]
    public string? SourceUrl { get; set; }

    public ApplicationStatus Status { get; set; }

    public decimal? SalaryMin { get; set; }
    public decimal? SalaryMax { get; set; }
    public DateTimeOffset? AppliedAt { get; set; }
    public DateTimeOffset? FollowUpDate { get; set; }
    public string? RecruiterName { get; set; }
    public string? RecruiterContact { get; set; }
    public string? SentCvVersion { get; set; }
    public string? RejectionReason { get; set; }

    public ICollection<JobOfferNote> Notes { get; set; } = [];
    public ICollection<JobOfferTechnology> RequiredTechnologies { get; set; } = [];

    [NotMapped] public List<int> RequiredSkillIds { get; set; } = [];
    [NotMapped] public List<string> RequiredSkillNames { get; set; } = [];
}